ListingManager/ListingManager.cs

```diff
@@ -318,38 +204,79 @@ public static bool GetPathToAccompanyingUnitTest(string listingPath, out string
-        // thanks to this we will pass everything as one command
-        command = command.Replace("\"", "\"\"");
-
-        string fileName = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
-            ? "CMD.exe"
-            : "/bin/bash";
-
-        var proc = new Process
-        {
-            StartInfo = new ProcessStartInfo
+    }
+
+    public static List<ListingInformation> PopulateListingDataFromPath(string pathToChapter, bool singleDir)
+    {
+        List<ListingInformation> listingData = new();
+        List<ListingInformation> testListingData = new();
+        var listingFiles = FileManager.GetAllFilesAtPath(pathToChapter)
+            .OrderBy(x => x).ToList();
+        foreach (string fileName in listingFiles)
+        {
+            bool result = TryGetListing(fileName, out ListingInformation? data);
+            if (result)
             {
-                FileName = fileName,
-                Arguments = "-c \"" + command + "\"",
-                UseShellExecute = false,
-                RedirectStandardOutput = true,
-                CreateNoWindow = true
+                if (data is not null)
+                {
+                    if (data.IsTest)
+                    {
+                        testListingData.Add(data);
+                    }
+                    else
+                    {
+                        listingData.Add(data);
+                    }
+                }
+                else
+                {
+                    throw new InvalidOperationException("Listing data is unexpectedly null with a successful result");
+                }
+            }
+        }
+
+        if (!singleDir)
+        {
+            var listingTestFiles = FileManager.GetAllFilesAtPath($"{pathToChapter}.Tests")
+    .OrderBy(x => x).ToList();
+            foreach (string fileName in listingTestFiles)
+            {
+                bool result = TryGetListing(fileName, out ListingInformation? data, true);
+                if (result)
+                {
+                    if (data is not null)
+                    {
+                        testListingData.Add(data);
+                    }
+                    else
+                    {
+                        throw new InvalidOperationException("Listing data is unexpectedly null with a successful result");
+                    }
+                }
             }
-        };
+        }
+        foreach (ListingInformation testListingInformation in testListingData)
+        {
+            ListingInformation listingInformation = listingData.Where(x => x.OriginalListingNumber == testListingInformation.OriginalListingNumber && x.OriginalChapterNumber == testListingInformation.OriginalChapterNumber && x.OriginalListingNumberSuffix == testListingInformation.OriginalListingNumberSuffix).First();
+            if (testListingInformation.Caption.ToLowerInvariant() == "Tests".ToLowerInvariant() && listingInformation.Caption != string.Empty)

```

Comment:
You don't want to do case-insensitive comparisons like this. A better option is to use the overloads on `string.Equals` that let you specify that you don't care about case. Doing it this way allocates extra strings that are not needed.